generate/generate.go

```diff
@@ -59,44 +60,67 @@ func BuildStructs(schemaPath string, outputDir string) error {
 		name := strings.Split(filepath.Base(allOfPath), ".")[0]
 		embeds = append(embeds, exportedName(name))
 
-		if err := buildStructFile(schemaPath, allOfPath, name, packageName, nil, outputDir); err != nil {
-			return fmt.Errorf("failed to build struct file for %q: %v", name, err)
+		if err := buildFiles(schemaPath, allOfPath, name, packageName, nil, outputDir, useMessagePack); err != nil {
+			return err
 		}
 	}
 
 	for _, oneOfPath := range oneOfTypes {
 		name := strings.Split(filepath.Base(oneOfPath), ".")[0]
 
-		if err := buildStructFile(schemaPath, oneOfPath, name, packageName, embeds, outputDir); err != nil {
-			return fmt.Errorf("failed to build struct file for %q: %v", name, err)
+		if err := buildFiles(schemaPath, oneOfPath, name, packageName, embeds, outputDir, useMessagePack); err != nil {
+			return err
 		}
 	}
 
 	return nil
 }
 
-func buildStructFile(schemaPath, childPath, name, packageName string, embeds []string, outputDir string) error {
+// Build specified struct file and (if enabled) MessagePack serialization methods.
+func buildFiles(schemaPath, childPath, name, packageName string, embeds []string, outputDir string, useMessagePack bool) error {
+	outPath, err := buildStructFile(schemaPath, childPath, name, packageName, embeds, outputDir)
+	if err != nil {
+		return fmt.Errorf("failed to build struct file for %q: %v", name, err)
+	}
+
+	if useMessagePack {
+		msgp := "msgp"
+
+		if gopath := os.Getenv("GOPATH"); gopath != "" {
```

Comment:
This assumes that msgp is in the GOPATH but I don't think that is a good assumption. Rather than requiring it be installed everywhere I would consider explicitly adding it to Gopkg.toml and then calling it via `go run` in the vendor directory, then fall back to looking for it in the standard $PATH. That means you don't have to install anything to use this when you have the code checked out and automatically get the right version. That is how we often use the package. If you have built jstransform and distributed as a binary you will need msgp as well, the failure to find the executable should be called out as a specific error so the user knows they must install it.